lightning/src/routing/router.rs

```diff
@@ -495,6 +495,13 @@ impl<'a> PaymentPath<'a> {
 		self.hops.last().unwrap().0.fee_msat
 	}
 
+	fn get_path_penalty_msat(&self) -> u64 {
+		if self.hops.len() < 1 {
+			return u64::max_value();
+		}
+		self.hops.first().unwrap().0.path_penalty_msat
```

Comment:
More of a personal preference here but consider one-lining this using `self.hops.first().map(..).unwrap_or(..)` instead of checking the length. Feel free to ignore, of course.

Reply:
Sure looks cleaner! Addressed with https://github.com/lightningdevkit/rust-lightning/pull/1359/commits/d5a07380ebd653f44d245ce4516a9dcf8332cfd3.